api/comments.rb

```diff
@@ -27,16 +27,34 @@
   sub_comment.anonymous_to_peers = bool_anonymous_to_peers || false
   sub_comment.author = user
   sub_comment.comment_thread = comment.comment_thread
+  sub_comment.children_count = 0
   sub_comment.save
   if sub_comment.errors.any?
     error 400, sub_comment.errors.full_messages.to_json
   else
-    user.subscribe(comment.comment_thread) if bool_auto_subscribe
-    sub_comment.to_hash.to_json
+    comment.set :children_count, comment.children_count + 1
```

Comment:
I'm still wondering if we can get what you want without caching this because caching is tricky.  This code has a potential race condition where 2 comments are added at once, the original count is 0 and both try to set it to 0 + 1 = 1, and the final count is 1 instead of 2.
  
If we get the count via a count query and cache that value, we still might be off, but it would be self-correcting with later comments.

Again - I'm going to continue to think about an alternative to caching.